tests/core/evaluation/test_markers_config.py

```diff
@@ -104,17 +105,74 @@ def test_load_invalid_path():
         MarkerConfig.load_config_from_path("not a path")
 
 
-def test_load_valid_file(simple_marker_config: Text, simple_marker_config_json):
+def test_load_valid_file(simple_markers_config, simple_marker_config_json):
     """Tests the single config loader"""
-    yaml_as_dict = MarkerConfig.load_config_from_path(simple_marker_config)
+    yaml_as_dict = MarkerConfig.load_config_from_path(simple_markers_config)
     assert yaml_as_dict == simple_marker_config_json
 
 
-def test_load_valid_path(multi_marker_config_folder, multi_marker_config_json):
+def test_load_valid_path(markers_config_folder, multi_marker_config_json):
     """Tests the config folder loading"""
-    yaml_as_dict = MarkerConfig.load_config_from_path(multi_marker_config_folder)
+    yaml_as_dict = MarkerConfig.load_config_from_path(markers_config_folder)
     # check that the two configs contain the same entries.
     for m in yaml_as_dict["markers"]:
         assert m in multi_marker_config_json["markers"]
     for m in multi_marker_config_json["markers"]:
         assert m in yaml_as_dict["markers"]
+
+
+def test_valid_config(simple_marker_config_json):
+    assert MarkerConfig._validate_config(simple_marker_config_json) is True
+
+
+def test_valid_config_operators(markers_config_operators):
+    config = MarkerConfig.load_config_from_path(markers_config_operators)
+    assert MarkerConfig._validate_config(config) is True
+
+
+def test_config_missing_required_top_level_markers_label():
+    sample_json = {"marker": "no_restart"}
+    with pytest.raises(ValidationError):
+        MarkerConfig._validate_config(sample_json)
+
+
+def test_config_missing_required_marker_label():
+    sample_json = {
+        "markers": [{"condition": [{"action_not_executed": ["action_restart"]}]}]
+    }
+    with pytest.raises(ValidationError):
+        MarkerConfig._validate_config(sample_json)
+
+
+def test_config_missing_required_condition_label():
+    sample_json = {"markers": [{"marker": "no_restart"}]}
+    with pytest.raises(ValidationError):
+        MarkerConfig._validate_config(sample_json)
+
+
+def test_config_invalid_operator():
+    sample_json = {
+        "markers": [
+            {
+                "marker": "no_restart",
+                "operator": "XOR",
+                "condition": [{"action_not_executed": ["action_restart"]}],
+            },
+        ]
+    }
+    with pytest.raises(ValidationError):
+        MarkerConfig._validate_config(sample_json)
+
+
+def test_config_invalid_event():
+    sample_json = {
```

Comment:
if you change to using `from_yaml`, you'll also have to change to using yaml and that will make the tests more readable in my opinion.

Reply:
I find Json easier to work with 🙈 yaml formatting drives me crazy.

Reply:
Pro-tip: You could use `textwrap.dedent(<yaml-content>)` to get rid of some of the whitespace related issues.

Reply:
Cool!